### DISCUSSION TITLE
vulnerability to XSS

### DISCUSSION THREAD

**Discussion by rbartczak** [nilsteampassnet/TeamPass#3653] [Q&A]
We have received information from our security team about an XSS vulnerability (e.g. session stealing).
I can see in the code that there is information about XSS security.

Have you tested xss vulnerabilities?

**Comment by nilsteampassnet** [comment]
It is regularly tested by external as I receive possible CVEs details.
Most of time they are related to version 2.x which is not maintened anymore.
Very few of them concerns version 3.x for which security is taken into account from design.
I then solves them once fix identified.
But personnaly I don't perform such tests.

But please provide me what your security team shared with you.

**Reply by rbartczak** [reply]
Hello,

Is it possible to add xss filtering (e.g. xss_clean) for fields that are displayed to other users (username, email address, item name, etc)

**Reply by nilsteampassnet** [reply]
All fields are already sanitized using [Elegantweb/sanitize](https://github.com/elegantweb/sanitizer) library which relies on PHP htmlspecialchars() function.

**Reply by rbartczak** [reply]
Hello,

Thank you for your answer. Maybe I need to enable something, install some server-side extension?
For me, I can enter xss/scripts in the fields
![tp_xss](https://user-images.githubusercontent.com/48474593/232407244-5655a000-c145-4922-bd2e-aeecb0f84b89.jpg)

**Comment by gamblerpentest** [comment]
I'm bumping and confirm issue with this topic. In the text field where the user can change the name/surname, is still an error that allows the execution of the XSS script (confirming what rbartczak wrote).
Applies to version 3.x.

**Reply by rbartczak** [reply]
Hello,

Thank you very much for implementing DOMPurify, great job!

Is it possible to include the Label/Login field when creating a new item?

**Reply by rbartczak** [reply]
Same request in folder name/label.
really thanks a lot for your support

**Comment by nilsteampassnet** [comment]
@rbartczak 
it is the case already. I have added DOMPurify on all text fields.

![image](https://github.com/nilsteampassnet/TeamPass/assets/1197546/40e8ff4a-4268-451d-88cc-d856c1572f09)

**Comment by Saibamen** [comment]
Please retest in 3.0.7 and close if fixed
